pandas/tests/scalar/test_na_scalar.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+import pytest
+
+from pandas.core.na_scalar import NA
+
+
+def test_singleton():
+    assert NA is NA
+    new_NA = type(NA)()
+    assert new_NA is NA
+
+
+def test_repr():
+    assert repr(NA) == "NA"
+    assert str(NA) == "NA"
+
+
+def test_truthiness():
+    with pytest.raises(TypeError):
+        bool(NA)
+
+    with pytest.raises(TypeError):
+        not NA
+
+
+def test_hashable():
+    assert hash(NA) == hash(NA)
+    d = {NA: "test"}
+    assert d[NA] == "test"
+
+
+def test_arithmetic_ops(all_arithmetic_functions):
+    op = all_arithmetic_functions
+
+    for other in [NA, 1, 1.0, "a", np.int64(1)]:
+        if op.__name__ == "rmod" and isinstance(other, str):
+            continue
+        assert op(NA, other) is NA
+
+
+def test_comparison_ops():
+
+    for other in [NA, 1, 1.0, "a", np.int64(1)]:
+        assert (NA == other) is NA
+        assert (NA != other) is NA
+        assert (NA > other) is NA
+        assert (NA >= other) is NA
+        assert (NA < other) is NA
+        assert (NA <= other) is NA
+
+        if isinstance(other, np.int64):
+            # for numpy scalars we get a deprecation warning and False as result
+            # for equality or error for larger/lesser than
+            continue
```

Comment:
So numpy scalars we don't have full control over, so this means that if they are the left operand, we get some other behaviour:

```
In [27]: np.int64(1) == pd.NA 
/home/joris/miniconda3/envs/dev/bin/ipython:1: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
  #!/home/joris/miniconda3/envs/dev/bin/python
Out[27]: False

In [28]: pd.NA == np.int64(1) 
Out[28]: NA

In [29]: np.int64(1) < pd.NA 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-29-87134fac2734> in <module>
----> 1 np.int64(1) < pd.NA

~/scipy/pandas/pandas/core/na_scalar.py in __bool__(self)
     37 
     38     def __bool__(self):
---> 39         raise TypeError("boolean value of NA is ambiguous")
     40 
     41     def __hash__(self):

TypeError: boolean value of NA is ambiguous

In [30]: pd.NA > np.int64(1) 
Out[30]: NA
```

(for the first case, not sure what the behaviour will be once the change in numpy is done)